common/os_calls.c

```diff
@@ -416,11 +416,27 @@ g_tcp_socket(void)
 
 #if defined(XRDP_ENABLE_IPV6)
     rv = (int)socket(AF_INET6, SOCK_STREAM, 0);
+    if (rv < 0)
+    {
+        log_message(LOG_LEVEL_ERROR, "g_tcp_socket: %s", strerror(errno));
+
+        switch(errno)
+        {
+            case EAFNOSUPPORT: /* if IPv6 not supported, retry IPv4 */
+                log_message(LOG_LEVEL_INFO, "IPv6 not supported, falling back to IPv4");
```

Comment:
I have some doubts about it. Should the caller of g_tcp_socket() decide whether to use IPv6?